lib/travis/build/script/r.rb

```diff
@@ -0,0 +1,368 @@
+            sh.cmd 'sudo apt-key adv --keyserver keyserver.ubuntu.com ' +
+                   '--recv-keys E084DAB9'
+
+            # Add marutter's c2d4u repository.
+            sh.cmd 'sudo add-apt-repository -y "ppa:marutter/rrutter"'
+            sh.cmd 'sudo add-apt-repository -y "ppa:marutter/c2d4u"'
+
+            # Update after adding all repositories. Retry several
+            # times to work around flaky connection to Launchpad PPAs.
+            sh.cmd 'sudo apt-get update -qq', retry: true
+
+            # Install an R development environment. qpdf is also needed for
+            # --as-cran checks:
+            #   https://stat.ethz.ch/pipermail/r-help//2012-September/335676.html
+            sh.cmd 'sudo apt-get install --no-install-recommends r-base-dev ' +
+                   'r-recommended qpdf', retry: true
+
+            # Change permissions for /usr/local/lib/R/site-library
+            # This should really be via 'sudo adduser travis staff'
+            # but that may affect only the next shell
+            sh.cmd 'sudo chmod 2777 /usr/local/lib/R /usr/local/lib/R/site-library'
+
+          when 'osx'
+            # Install from latest CRAN binary build for OS X
+            sh.cmd "wget #{config[:cran]}/bin/macosx/R-latest.pkg " +
+                   '-O /tmp/R-latest.pkg'
+
+            sh.echo 'Installing OS X binary package for R'
+            sh.cmd 'sudo installer -pkg "/tmp/R-latest.pkg" -target /'
+            sh.rm '/tmp/R-latest.pkg'
+
+          else
+            sh.failure "Operating system not supported: #{config[:os]}"
+          end
+
+          if needs_bioc?
+            setup_bioc
+          end
+          if config[:latex]
+            setup_latex
+          end
+          if config[:pandoc]
+            setup_pandoc
+          end
+        end
+
+        def announce
+          super
+
+          sh.cmd 'Rscript -e \'sessionInfo()\''
+          sh.echo ''
+        end
+
+        def install
+          super
+
+          # Install any declared packages
+          apt_install config[:apt_packages]
+          r_binary_install config[:r_binary_packages]
+          r_install config[:r_packages]
+          bioc_install config[:bioc_packages]
+          r_github_install config[:r_github_packages]
+
+          # Install dependencies for the package we're testing.
+          install_deps
+        end
+
+        def script
+          # Build the package
+          sh.echo "Building with: R CMD build ${R_BUILD_ARGS}"
+          sh.cmd "R CMD build #{config[:r_build_args]} ."
+          tarball_script = [
+            'library("devtools")',
+            'pkg <- as.package(".")',
+            'cat(paste0(pkg$package, "_", pkg$version, ".tar.gz"))',
+          ].join('; ')
+          sh.export 'PKG_TARBALL', "$(Rscript -e '#{tarball_script}')"
+
+          # Test the package
+          sh.echo 'Testing with: R CMD check "${PKG_TARBALL}" ' +
+                  "#{config[:r_check_args]}"
```

Comment:
i kept it on two lines to stay *vaguely* near 80 characters. i don't know that anyone has an 80 char limit, but working at google has apparently convinced me that long lines and trailing whitespace are sins of the highest order. (i think the conventional name here is "stockholm syndrome"?)